modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/select/SelectDDMFormFieldValueAccessor.java

```diff
@@ -82,11 +82,11 @@ public JSONArray getValueForEvaluation(
 				optionsValuesJSONArray.getString(i));
 
 			if (matcher.matches()) {
-				return createJSONArray(
-					StringBundler.concat(
-						StringPool.OPEN_BRACKET,
-						getOptionsLabels(ddmFormFieldValue, locale),
-						StringPool.CLOSE_BRACKET));
+				JSONArray valueJSONArray = jsonFactory.createJSONArray();
+
+				valueJSONArray.put(getOptionsLabels(ddmFormFieldValue, locale));
+
+				return valueJSONArray;
```

Comment:
Hi @antonio91dev, can you please give more details about your proposed fix?

Apparently, you just removed the brackets, am I right? If so, how is it fixing the reported issue?

Thanks in advance for your answers. :slightly_smiling_face:

Reply:
Hi @natocesarrego , The function must return a JSONARRAY, for which I am adding values to a json array so that the interpretation of the values entered to the arrays are correct.

Reply:
what the createJSONArray does is create an array based on a "string" here there are problems when the "string" are "/" ",".

Reply:
Thanks for your answer, @antonio91dev. :slightly_smiling_face: 

What error are you exactly getting? Is it _Unable to parse JSON array_?

Have you find out why these specific characters `/` and `,` can't be interpreted correctly?

Reply:
Thanks for your explanation, @antonio91dev. :slightly_smiling_face: 

I just have one suggestion regarding your proposed fix. Can we create the `JSONArray` using the existing method `createJSONArray` instead? The benefit is catch thrown exceptions and also log it, please see the code below:
```
JSONArray jsonArray = createJSONArray("[]");

jsonArray.put(getOptionsLabels(ddmFormFieldValue, locale));

return jsonArray;
```

Reply:
okay @natocesarrego  , I am going to proceed to create a method to be able to detect exceptions.

Reply:
Hey @antonio91dev, the method `createJSONArray` already exists in the class `SelectDDMFormFieldValueAccessor.java`. :wink: You just need to use it. :slightly_smiling_face:

Reply:
@natocesarrego  , I understand, I'm going to try and let you know the result

Reply:
Hi @natocesarrego  the change is already in my branch